tests/rptest/clients/rpk.py

```diff
@@ -382,7 +382,7 @@ def sasl_allow_principal(self,
             ",".join(operations), resource, resource_name, "--brokers",
             self._redpanda.brokers(), "--user", username, "--password",
             password, "--sasl-mechanism", mechanism
-        ]
+        ] + self._kafka_conn_settings()
```

Comment:
It looks like `_kafka_conn_settings` includes a bunch of stuff such as username and password, so this could probably do with some reworking.